### PR TITLE
Replace luet-makeiso with elemental build-iso.

### PR DESCRIPTION
**Pull request by masteryyh**
**Problem:**

We need to migrate from `luet-makeiso` to `elemental build-iso`.

**Solution:**

Add `elemental` base image for elemental binary and `rsync` package for elemental `build-iso` command execution.

**Related Issue:**
https://github.com/harvester/harvester/issues/2204

**Test plan:**

Boot:
1. Build an ISO from `elemental-build` branch (both `harvester` and `harvester-installer` repo);
2. Try to boot up using this ISO in UEFI environment, it should boot up without problem;
3. Try to boot up using this ISO in legacy BIOS environment too.

Upgrade:
1. Build an ISO from `elemental-build` branch (both `harvester` and `harvester-installer` repo);
2. Install a clean environment using v1.1.2;
3. Create a http server to serve the ISO file;
4. Create a version object in Harvester and let Harvester upgrade to the ISO just built;
5. Check if Harvester upgraded successfully.

**Additional Information:**
We might need to create another issue to add legacy BIOS boot capability if it's necessary.

### CODE REVIEW

File: Dockerfile.dapper [hunk 1]
```diff
@@ -1,5 +1,6 @@
 FROM quay.io/costoolkit/releases-green:luet-toolchain-0.21.2 as luet
 FROM quay.io/costoolkit/releases-green:luet-makeiso-toolchain-0.4.0 as makeiso
```
Comment (bk201) [inline Dockerfile.dapper:2]: We can remove line 2 (luet-makeiso) if it's not needed anymore.

File: package/harvester-os/iso/boot/grub2/grub.cfg [hunk 2]
```diff
@@ -22,17 +22,17 @@ if [ -f ${font} ];then
 fi
 menuentry "Harvester Installer ${harvester_version}" --class os --unrestricted {
     echo Loading kernel...
-    $linux ($root)/boot/kernel.xz cdroot root=live:CDLABEL=COS_LIVE rd.live.dir=/ rd.live.squashimg=rootfs.squashfs console=tty1 console=ttyS0 rd.cos.disable net.ifnames=1
+    $linux ($root)/boot/kernel cdroot root=live:CDLABEL=COS_LIVE rd.live.dir=/ rd.live.squashimg=rootfs.squashfs console=tty1 console=ttyS0 rd.cos.disable net.ifnames=1
     echo Loading initrd...
-    $initrd ($root)/boot/rootfs.xz
+    $initrd ($root)/boot/initrd
```
Comment (guangbochen) [inline package/harvester-os/iso/boot/grub2/grub.cfg:27]: is there any reference for those changes?

File: Dockerfile.dapper [hunk 3]
```diff
@@ -33,6 +34,7 @@ RUN mkdir /usr/tmp && \
 # luet & makeiso
 COPY --from=luet /usr/bin/luet /usr/bin/luet
 COPY --from=makeiso /usr/bin/luet-makeiso /usr/bin/luet-makeiso
```
Comment (Vicente-Cheng) [inline Dockerfile.dapper:36]: If we remove line 2, please also remove this line.

**Comment by masteryyh** [comment]
As I'm working on adding legacy BIOS support, converting this PR to draft.

File: scripts/package-harvester-os [hunk 4]
```diff
@@ -67,13 +83,54 @@ find $BUNDLE_DIR -type f -exec chmod +r {} +
...
+elemental build-iso --config-dir "$(pwd)" "docker:${HARVESTER_OS_IMAGE}" -a x86_64 \
+          --local \
+          -n "${ISO_PREFIX}" \
+          -o "${ARTIFACTS_DIR}" \
+          --overlay-iso "$(pwd)/iso" \
+          -x "-comp xz"
+
+rm -f ${ARTIFACTS_DIR}/${ISO_PREFIX}.iso.sha256
+
+# Unpack ISO for reconfiguring boot image
```
Comment (bk201) [inline scripts/package-harvester-os:97]: @masteryyh Please add more comments on why we need to unpack and pack the ISO again. Thanks!

File: scripts/package-harvester-os [hunk 5]
```diff
@@ -67,13 +83,54 @@ find $BUNDLE_DIR -type f -exec chmod +r {} +
...
+extract_dir=$(mktemp -d)
+xorriso -osirrox on -indev ${ARTIFACTS_DIR}/${ISO_PREFIX}.iso -extract / "${extract_dir}"
+traverse_set_permission "${extract_dir}"
```
Comment (starbops) [inline scripts/package-harvester-os:100]: Is it possible to replace this function and its relevant recursive calls with the following command?

```
chmod -R a=r,u+w,a+X "${extract_dir}"
```

**Review by starbops — APPROVED** [review]
Tested upgrade from v1.1.2 to this PR with legacy BIOS and UEFI. LGTM, thanks!

**Review by Vicente-Cheng — APPROVED** [review]
Tested with the upgrade with legacy mode. All basic operations and downstream cluster verify look fine. LGTM, thanks!

**Review by bk201 — APPROVED** [review]
LGTM

**Pull request merged**